mono/metadata/handle.h

```diff
@@ -32,15 +24,33 @@ typedef MonoHandleStorage* MonoHandle;
  * The field obj is not private as there is no way to do that
  * in C, but using a C++ template would simplify that a lot
  */
-struct _MonoHandleStorage {
-	MonoObject *obj;
-};
+typedef struct {
+	MonoObject *__private_obj;
+} MonoHandleStorage;
+
+typedef MonoHandleStorage* MonoHandle;
+
+#include "handle-arena.h"
```

Comment:
Up to you. I don't have a strong preference either way.

(My weak preference is for "headers in a separate 'include' directory with nested subdirectories, one thing per header, roll-up headers for client convenience in base directories", but that's not what Mono looks like, generally, so it's not particularly relevant)